addons/mail/static/src/components/chatter_topbar/chatter_topbar.xml

```diff
@@ -36,14 +36,22 @@
                         </button>
                     </t>
                     <t t-if="chatterTopbar.chatter.hasActivities">
-                        <button class="o_ChatterTopbar_button o_ChatterTopbar_buttonScheduleActivity btn btn-link" type="button" t-att-disabled="chatterTopbar.chatter.isDisabled" t-on-click="chatterTopbar.chatter.onClickScheduleActivity" title="Schedule an activity" data-hotkey="shift+a">
+                        <button class="o_ChatterTopbar_button o_ChatterTopbar_buttonScheduleActivity btn btn-link" type="button" t-att-disabled="!chatterTopbar.chatter.hasWriteAccess" t-on-click="chatterTopbar.chatter.onClickScheduleActivity" title="Schedule an activity" data-hotkey="shift+a">
                             <i class="fa fa-clock-o"/>
                             Schedule activity
                         </button>
                     </t>
                     <div class="flex-grow-1"/>
                         <div class="o_ChatterTopbar_rightSection flex-grow-1 flex-shrink-0 justify-content-end d-flex">
-                            <button class="o_ChatterTopbar_button o_ChatterTopbar_buttonAttachments btn btn-link" type="button" t-att-disabled="chatterTopbar.chatter.isDisabled" t-att-aria-expanded="chatterTopbar.chatter.attachmentBoxView ? 'true' : 'false'" t-on-click="chatterTopbar.chatter.onClickButtonAttachments" title="Show Attachments">
+
+                            <button t-if="chatterTopbar.chatter.thread.allAttachments.length === 0" class="o_ChatterTopbar_button o_ChatterTopbar_buttonAddAttachments btn btn-link" type="button" t-att-disabled="!chatterTopbar.chatter.hasWriteAccess" t-att-aria-expanded="chatterTopbar.chatter.attachmentBoxView ? 'true' : 'false'" t-on-click="chatterTopbar.chatter.onClickButtonAddAttachments" title="Add Attachments">
+                                <i class="fa fa-paperclip" role="img" aria-label="Attachments"/>
+                                Attach files
+                                <t t-if="chatterTopbar.chatter.isShowingAttachmentsLoading">
+                                    <i class="o_ChatterTopbar_buttonAttachmentsCountLoader fa fa-circle-o-notch fa-spin ms-1" aria-label="Attachment counter loading..."/>
+                                </t>
```

Comment:
On the other button the spinner is supposed to be replacing the text while they are loading, shouldn't it be the case here too?

Reply:
Well this is the annoying part. Because `isShowingAttachmentsLoading` is in fact related to `/mail/thread/data` and not really to attachment. We can't know how much attachment we have on the thread before the thread is loaded (an the spinner not needed anymore).